config/opal_setup_wrappers.m4

```diff
@@ -333,146 +394,270 @@ AC_DEFUN([OPAL_SETUP_WRAPPER_FINAL],[
+              OPAL_FLAGS_APPEND_UNIQ([OMPI_WRAPPER_LDFLAGS_STATIC], [${ompi_mca_wrapper_extra_static_ldflags} ${wrapper_extra_static_ldflags}])
+              OPAL_FLAGS_APPEND_MOVE([OMPI_WRAPPER_LIBS_STATIC], [${ompi_mca_wrapper_extra_static_libs} ${wrapper_extra_static_libs}])])
+
+       dnl Add the user-provided flags
+       OPAL_FLAGS_APPEND_UNIQ([OMPI_WRAPPER_LDFLAGS], [${with_wrapper_ldflags}])
+       OPAL_FLAGS_APPEND_MOVE([OMPI_WRAPPER_LIBS], [${with_wrapper_libs}])
+
+       dnl fortran FTW!
+       OMPI_WRAPPER_FC_LIBS="${OMPI_FORTRAN_USEMPIF08_LIB} ${OMPI_FORTRAN_USEMPI_LIB} ${OMPI_FORTRAN_MPIFH_LINK} ${OMPI_WRAPPER_LIBS}"
+       OMPI_WRAPPER_FC_LIBS_STATIC=${OMPI_WRAPPER_LIBS_STATIC}
+       OMPI_WRAPPER_FC_LDFLAGS=$OMPI_WRAPPER_LDFLAGS
+       OMPI_WRAPPER_FC_LDFLAGS_STATIC=$OMPI_WRAPPER_LDFLAGS
+
+       RPATHIFY_LDFLAGS([OMPI_WRAPPER_LDFLAGS])
+       RPATHIFY_LDFLAGS([OMPI_WRAPPER_LDFLAGS_STATIC])
+       RPATHIFY_FC_LDFLAGS([OMPI_WRAPPER_FC_LDFLAGS])
+       RPATHIFY_FC_LDFLAGS([OMPI_WRAPPER_FC_LDFLAGS_STATIC])
+
+       OPAL_FLAGS_APPEND_UNIQ([OMPI_WRAPPER_LDFLAGS], [${runpath_args}])
+       OPAL_FLAGS_APPEND_UNIQ([OMPI_WRAPPER_FC_LDFLAGS], [${runpath_fc_args}])
+
+       AC_MSG_CHECKING([for OMPI wrapper LDFLAGS])
+       AC_SUBST([OMPI_WRAPPER_LDFLAGS])
+       AC_MSG_RESULT([$OMPI_WRAPPER_LDFLAGS])
+
+       AC_MSG_CHECKING([for OMPI wrapper static LDFLAGS])
+       AC_SUBST([OMPI_WRAPPER_LDFLAGS_STATIC])
+       AC_MSG_RESULT([$OMPI_WRAPPER_LDFLAGS_STATIC])
+
+       AC_MSG_CHECKING([for OMPI wrapper Fortran LDFLAGS])
+       AC_SUBST([OMPI_WRAPPER_LDFLAGS])
+       AC_MSG_RESULT([$OMPI_WRAPPER_LDFLAGS])
+
+       AC_MSG_CHECKING([for OMPI wrapper Fortran static LDFLAGS])
+       AC_SUBST([OMPI_WRAPPER_LDFLAGS_STATIC])
+       AC_MSG_RESULT([$OMPI_WRAPPER_LDFLAGS_STATIC])
+
+       AC_MSG_CHECKING([for OMPI wrapper LIBS])
+       AC_SUBST([OMPI_WRAPPER_LIBS])
+       AC_MSG_RESULT([$OMPI_WRAPPER_LIBS])
+
+       AC_MSG_CHECKING([for OMPI wrapper static LIBS])
+       AC_SUBST([OMPI_WRAPPER_LIBS_STATIC])
+       AC_MSG_RESULT([$OMPI_WRAPPER_LIBS_STATIC])
+
+       AC_MSG_CHECKING([for OMPI wrapper Fortran LDFLAGS])
+       AC_SUBST([OMPI_WRAPPER_FC_LDFLAGS])
+       AC_MSG_RESULT([$OMPI_WRAPPER_FC_LDFLAGS])
+
+       AC_MSG_CHECKING([for OMPI wrapper Fortran static LDFLAGS])
+       AC_SUBST([OMPI_WRAPPER_FC_LDFLAGS_STATIC])
+       AC_MSG_RESULT([$OMPI_WRAPPER_LDFLAGS_STATIC])
+
+       AC_MSG_CHECKING([for OMPI wrapper Fortran LIBS])
+       AC_SUBST([OMPI_WRAPPER_FC_LIBS])
+       AC_MSG_RESULT([$OMPI_WRAPPER_FC_LIBS])
+
+       AC_MSG_CHECKING([for OMPI wrapper Fortran static LIBS])
+       AC_SUBST([OMPI_WRAPPER_FC_LIBS_STATIC])
+       AC_MSG_RESULT([$OMPI_WRAPPER_FC_LIBS_STATIC])
+
+       # language binding support.  C++ is a bit different, as the
+       # compiler should work even if there is no MPI C++ bindings
+       # support.  However, we do want it to fail if there is no C++
```

Comment:
I thought we can build now without a c++ compiler on the system:
https://github.com/open-mpi/ompi/pull/9075

Reply:
"we" here is ambiguous (and this was a code move, so this comment isn't new).  Here "we" means the wrapper compiler, not configure.  So if there was no C++ compiler, OMPI would configure and build.  When someone invoked mpic++, the wrapper would spit out a "not supported" error, rather than a "binary not found" error message.